### PR TITLE
fix(samples): Updated and fixed the secrets sample

### PR DESCRIPTION
**Pull request by Ark-kun**
The sample started failing with "ImportError: cannot import name 'collections_abc' from 'six.moves' (unknown location)".
While fixing the issue I saw that the sample is pretty outdated (for example, we've deprecated the direct ContainerOp construction), so I've decided to update the sample along the way.

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/4585/kubeflow-pipeline-sample-test/1313222398857711616

### CODE REVIEW

File: samples/core/secret/secret.py [hunk 1]
```diff
@@ -18,35 +18,43 @@
...
+    buckets = storage_client.list_buckets()
+    print("List of buckets:")
+    for bucket in buckets:
+        print(bucket.name)
+
+
+gcs_list_buckets_op = kfp.components.create_component_from_func(
+    gcs_list_buckets,
+    base_image='python:3.7',
+    packages_to_install=['google-cloud-storage==1.31.2'],
```
Comment (chensun) [inline samples/core/secret/secret.py:56]: So real the fix is to use a higher version of `google-cloud-storage`? Can you explain a bit why do we need the other part of the change (`gcs_read_op` -> `gcs_list_items_op`)?

**Review by chensun — COMMENTED** [review]
/lgtm

Thanks for the fix, Alexey

File: samples/core/secret/secret.py (hunk 1 above)
Comment (Ark-kun) [reply samples/core/secret/secret.py:56]: You're correct that these changes are not fully related.

I saw that the sample is pretty outdated (for example, we've deprecated the direct `ContainerOp` construction), so I've decided to update the sample along the way.

>gcs_read_op -> gcs_list_items_op

The original code of `gcs_read_op` does not really read from GCS, it uses `gsutil ls` to list the items.

**Pull request merged**